src/bottom-bar/approve-button/approve-button.js

```diff
@@ -0,0 +1,68 @@
+import { useAlert } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import { Button } from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React, { useState } from 'react'
+import { useWorkflowContext } from '../../workflow-context/index.js'
+import { ApproveModal } from './approve-modal/index.js'
+import { useApproveData } from './use-approve-data.js'
+
+const ApproveButton = ({ disabled }) => {
+    // state
+    const [unexpectedError, setUnexpectedError] = useState(null)
+    const [showApproveModal, setShowApproveModal] = useState(false)
+    const { params, refresh } = useWorkflowContext()
+
+    // callbacks
+    const showApprovalDialog = () => setShowApproveModal(true)
+    const hideApprovalDialog = () => setShowApproveModal(false)
+    const onApprove = () => {
+        const { wf, pe, ou } = params
+        approveData({ wf, pe, ou })
+    }
+
+    // api
+    const { show: showApprovalSuccess } = useAlert(i18n.t('Approval saved'))
+    const [approveData, { loading, error: approveError }] = useApproveData({
+        onComplete: () => {
+            showApprovalSuccess()
+            refresh()
+            hideApprovalDialog()
+        },
+        onError: e => {
+            setUnexpectedError(e)
+            hideApprovalDialog()
```

Comment:
Since the error will be shown in the modal we don't want to call `hideApprovalDialog` in `onError`